src/plugins/intel_cpu/tests/functional/cmake/target_per_test.cmake

```diff
@@ -4,40 +4,227 @@
 
 #create targed with prefix TARGET_PREFIX for each test file in directory TEST_DIR
 function(create_target_per_test_for_directory TEST_DIR TARGET_PREFIX)
-#exclude every other test file inside directory
-  set(EXCLUDED_SOURCE_PATHS_FOR_TEST
-    ${TEST_DIR})
+  # exclude every other test file inside directory (we pass explicit OBJECT_FILES)
+  set(EXCLUDED_SOURCE_PATHS_FOR_TEST ${TEST_DIR})
 
-#list of object files required for each test
+  # list of object files required for each test
   set(REQUIRED_OBJECT_FILES
     ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instances/core_config.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instances/skip_tests_config.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/shared_tests_instances/set_device_name.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/utils/cpu_test_utils.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/utils/fusing_test_utils.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/utils/transformations/insert_fake_quantize.cpp
-    ${CMAKE_CURRENT_SOURCE_DIR}/utils/transformations/insert_requantize.cpp
-  )
-
-if(X86_64)
-    list(APPEND REQUIRED_OBJECT_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/utils/x64/filter_cpu_info.cpp)
-elseif(ARM OR AARCH64)
-    list(APPEND REQUIRED_OBJECT_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/utils/arm/filter_cpu_info.cpp)
-elseif(RISCV64)
-    list(APPEND REQUIRED_OBJECT_FILES
-    ${CMAKE_CURRENT_SOURCE_DIR}/utils/riscv64/filter_cpu_info.cpp)
-endif()
+    ${CMAKE_CURRENT_SOURCE_DIR}/utils/transformations/insert_requantize.cpp)
+
+  if(X86_64)
+    list(APPEND REQUIRED_OBJECT_FILES ${CMAKE_CURRENT_SOURCE_DIR}/utils/x64/filter_cpu_info.cpp)
+  elseif(ARM OR AARCH64)
+    list(APPEND REQUIRED_OBJECT_FILES ${CMAKE_CURRENT_SOURCE_DIR}/utils/arm/filter_cpu_info.cpp)
+  elseif(RISCV64)
+    list(APPEND REQUIRED_OBJECT_FILES ${CMAKE_CURRENT_SOURCE_DIR}/utils/riscv64/filter_cpu_info.cpp)
+  endif()
+
+  # Explicit lists for directories
+  set(SLT_TOP_LEVEL
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/adaptive_pooling.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/augru_cell.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/augru_sequence.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/batch_to_space.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/broadcast.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/bucketize.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/concat.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/convert_to_plugin_specific_node.cpp
+    ${CMAKE_CURRENT_SOURCE_DIR}/custom/single_layer_tests/ctc_greedy_decoder.cpp
```

Comment:
Can we use file lists from the inner cmakefiles instead to avoid duplication?
This would require to "aggregate" those list of files into multiple variables which I think is actually a good thing. This will allow us to create extra scopes like "ov_cpu_func_subgraph_tests", etc